src/D2L.CodeStyle.Analyzers/Immutability/ImmutabilityContext.cs

```diff
@@ -138,6 +161,10 @@ out diagnostic
 						return true;
 					}
 
+					if( m_conditionalTypeParameters.Contains( type as ITypeParameterSymbol ) ) {

```

Comment:
```suggestion
					if( type is ITypeParameterSymbol tp && m_conditionalTypeParameters.Contains( tp ) ) {
```

would be slightly safer right, like wouldn't need to independently verify that `null` didn't get into the list accidentally.

(aside: enabling the no-implicit-null stuff in the analyzers might be interesting...)